modules/react/icon/lib/Graphic.tsx

```diff
@@ -79,23 +89,62 @@ export const graphicStencil = createStencil({
   },
 });
 
+export const graphicImageStencil = createStencil({
+  vars: {
+    width: '',
+    height: '',
+  },
+  base: ({width, height}) => ({
+    width,
+    height,
+    '& [data-part="graphic-img"]': {
+      maxWidth: '100%',
+      maxHeight: '100%',
+    },
+  }),
+});
+
 export const Graphic = createComponent('span')({
   displayName: 'Graphic',
-  Component: ({grow, width, height, ...elemProps}: GraphicProps, ref, Element) => {
+  Component: (
+    {grow, width, height, src, shouldMirror, ...elemProps}: GraphicProps,
+    ref,
+    Element
+  ) => {
+    console.log(src.svg);
     return (
-      <Svg
-        type={CanvasIconTypes.Graphic}
-        as={Element}
-        ref={ref}
-        {...handleCsProp(
-          elemProps,
-          graphicStencil({
-            grow,
-            width: typeof width === 'number' ? px2rem(width) : width,
-            height: typeof height === 'number' ? px2rem(height) : height,
-          })
-        )}
-      />
+      <>
+        {src.url ? (
+          <Element
+            ref={ref}
+            {...handleCsProp(
+              elemProps,
+              graphicImageStencil({
+                width: number2rem(width),
+                height: number2rem(height),
+              })
+            )}
+          >
+            <img data-part="graphic-img" src={src.url} alt="" />
+          </Element>
+        ) : typeof src.svg === 'string' && src.svg ? (
+          <Svg
+            type={CanvasIconTypes.Graphic}
+            as={Element}
+            ref={ref}
+            shouldMirror={shouldMirror}
+            src={src as OldCanvasGraphic}
+            {...handleCsProp(
+              elemProps,
+              graphicStencil({
+                grow,
+                width: number2rem(width),
+                height: number2rem(height),
+              })
+            )}
+          />
+        ) : null}
+      </>
```

Comment:
You don't really need `<></>` wrapper here, each step has only one element. Also, you can remove null, by making this:
```
typeof src.svg === 'string' && src.svg
->
typeof src.svg === 'string' && !!src.svg
```